Compiler/src/ProgramBuilder.ml

```diff
@@ -0,0 +1,727 @@
+open Program_types
+
+type var = int32
+
+(* Keeps a mapping of ids to temps *)
+type temp_map_t = (string, var, Core.String.comparator_witness) Core.Map.t
+
+type inst = instruction
+
+(* Keeps a mapping of ids that are still visible in Javascipt, but that Fuzzilli considers out of scope.
+Used to determine if var should be loaded from scope, rather than accessed as a temp *)
+type var_scope_map_t = (string, string, Core.String.comparator_witness) Core.Map.t
+
+type tracker = 
```

Comment:
Nit: would it be nicer to call this `builder` instead of `tracker`? It seems like it is doing more than just tracking variables

Reply:
Yeah, I've been thinking the same. I'll switch it over.